File: library/src/main/java/cn/wandersnail/widget/viewpager/BasePagerAdapter.java
package cn.wandersnail.widget.viewpager;

import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import java.util.List;

/**
 * date: 2019/8/22 23:28
 * author: zengfansheng
 */
public class BasePagerAdapter extends PagerAdapter {
    private List<Pager> pagers;

    public BasePagerAdapter(@NonNull List<Pager> pagers) {
        this.pagers = pagers;
    }

    @Override
    public int getCount() {
        return pagers.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }

    @Override
    public int getItemPosition(@NonNull Object object) {
        return PagerAdapter.POSITION_NONE;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        View view = pagers.get(position).getContentView();
        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((View) object);
    }

    public interface Pager {
        View getContentView();
    }
}

File: library/src/main/java/com/snail/widget/viewpager/BasePagerAdapter.kt
package com.snail.widget.viewpager

import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter

/**
 * Created by zengfs on 2016/2/21.
 * 基本ViewPager的基类
 */
open class BasePagerAdapter(private val pagers: MutableList<out Pager>) : PagerAdapter() {

    override fun getCount(): Int {
        return pagers.size
    }

    override fun getItemPosition(any: Any): Int {
        return PagerAdapter.POSITION_NONE
    }

    override fun isViewFromObject(view: View, o: Any): Boolean {
        return view == o
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = pagers[position].getContentView()
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, any: Any) {
        container.removeView(any as View)
    }
    
    interface Pager {
        fun getContentView(): View
    }
}
